stdlib/public/core/Integers.swift.gyb

```diff
@@ -3022,6 +3025,26 @@ ${assignmentOperatorComment('&' + x.operator, True)}
 % end
 }
 
+extension FixedWidthInteger {
+  public static func _random<R: RandomNumberGenerator>(
+    using generator: inout R
+  ) -> Self {
+    if bitWidth <= UInt64.bitWidth {
+      return Self(truncatingIfNeeded: generator.next() as UInt64)
+    }
+
+    let (quotient, remainder) = bitWidth.quotientAndRemainder(
+      dividingBy: UInt64.bitWidth
+    )
+    var tmp: Self = 0
+    for i in 0 ..< quotient + remainder.signum() {
+      let next: UInt64 = generator.next()
+      tmp += Self(truncatingIfNeeded: next) &<< (UInt64.bitWidth * i)
```

Comment:
It might be worth opening jira / radar to track making this less spectacularly inefficient for bignums at some future point?

Reply:
I think adding required `BinaryInteger` initializers to initialize integers from a given word or sequence of words would resolve this. I filed https://bugs.swift.org/browse/SR-7648; we should write up a swift-evolution pitch/proposal at some point. (cc @moiseev)

(The code above is defined on `FixedWidthInteger`, which uses a similar shifting loop for its own generic integer conversions. Frustratingly, `FixedWidthInteger` is currently impossible to implement using only public API: it requires an underscored version of the single-word initializer, without providing a default implementation.)

Reply:
Thanks for writing that up!